docs/source/target-query.rst

```diff
@@ -0,0 +1,121 @@
+target-query
+------------
+
+The **target-query** tool is one of the most prominent tools of Dissect. It allows you to
+**query a forensic image**
+to extract useful information from it. The basic usage format is:
+
+.. code-block:: console
+
+    $ target-query <TARGET> -f <FUNCTION_NAME>
+
+Here <TARGET> is the file that contains the forensic image and <FUNCTION_NAME> is the name
+of the function you wish to apply.
+
+Loading
+~~~~~~~
+
+Dissect attempts to determine the file format automatically and attempts to select an appropriate loader.
+If the image is split into multiple files, just feed it the first.
+
+If you wish to override auto-detection, use the ``-L`` option to explicitly specify the loader that
+has to be used:
+
+.. code-block:: console
+
+    $ target-query host123.vdi -L vbox
+
+.. note ::
+
+    The full list of loaders is listed at the end of the output of the ``-l`` option.
+
+
+In case there is no complete image available but just a couple of separate (log/evt/evtx) files, you can use
+the LogLoader for example:
+
+.. code-block:: console
+
+    $ target-query data/*.evtx -L log -f evtx
+
+It is also possible to load multiple separate disks, just string them together using the ``+`` character.
+
+.. code-block:: console
+
+    $ target-query disk1+disk2+disk3 -f osinfo
+    
+For encrypted disks like LUKS you can provide a key through the ``-K`` option to provide a keychain file
+or ``-Kv`` to provide a passphrase.
+
+For more information on the ``-K``, ``--keychain-file`` and ``-Kv``, ``--keychain-value`` arguments, please refer to
+:doc:`/usage/disk-encryption`.
+
+    
+Querying
+~~~~~~~~
+
+To get a full list of all functions available in target-query use the ``-l`` option. If you provide a target image
+the list will be filtered based on the compatibility with the target.
+
+You can apply multiple functions if you want:
+
+.. code-block:: console
+
+    $ target-query host.img -f runkeys,users
+    
+.. note ::
+
+    You can only combine functions that have the same output type.
+    If you don't, only the records will be shown.
+    
+If you have a lot of functions you wish to apply, wildcards (and other glob-rules) can be used:
+
+.. code-block:: console
+
+    $ target-query host.img -f apps.browser.*.history
+    
+You can combine the ``-f`` option with the `dry-run` option (``-n``/``--dry-run``) to see what will be actually executed:
```

Comment:
Whilst this parameter is definitely useful. I wouldn't expect someone who is new to dissect to use it. I would say it is more advanced use of target-query.

Reply:
It gives people a tool to explore the use of wildcards, so I think that is useful. Also, having a `--dry-run` option encourages people to play a bit, which is good. If the make a mistake, they don't have to wait a long time (because they run the wrong plugins) and they do not see a lot of useless errors. So this is a valuable tool for a beginner and a good step towards understanding the tool better.